src/main/java/org/cascadebot/cascadebot/events/ModlogEventListener.java

```diff
@@ -0,0 +1,1085 @@
+                    String current = getCurrentPermissionStateForHolder(permission, event.getChannel(), event.getPermissionHolder());
+                    if (!current.equals("x")) {
+                        stringBuilder.append("x > ").append(current).append(": ").append(permission.getName()).append('\n');
+                    }
+                }
+                for (Permission permission : ((PermissionOverrideUpdateEvent) event).getOldInherited()) {
+                    String current = getCurrentPermissionStateForHolder(permission, event.getChannel(), event.getPermissionHolder());
+                    if (!current.equals("\\\\")) {
+                        stringBuilder.append("\\ > ").append(current).append(": ").append(permission.getName()).append('\n');
+                    }
+                }
+                for (Permission permission : ((PermissionOverrideUpdateEvent) event).getOldAllow()) {
+                    String current = getCurrentPermissionStateForHolder(permission, event.getChannel(), event.getPermissionHolder());
+                    if (!current.equals("+")) {
+                        stringBuilder.append("+ > ").append(current).append(": ").append(permission.getName()).append('\n');
+                    }
+                }
+                embedFieldList.add(new ModlogEmbedField(false, "modlog.channel.perm.changed", "modlog.general.variable", stringBuilder.toString()));
+            } else {
+                return;
+            }
+            ModlogEventData modlogEventData = new ModlogEventData(modlogEvent, responsible, event.getChannel(), embedFieldList);
+            String holderMention = "";
+            if (event.getPermissionHolder() instanceof User) {
+                holderMention = ((User) event.getPermissionHolder()).getAsMention();
+            } else if (event.getPermissionHolder() instanceof Role) {
+                holderMention = ((Role) event.getPermissionHolder()).getAsMention();
+            }
+            modlogEventData.setExtraDescriptionInfo(List.of(holderMention));
+            guildData.getModeration().sendModlogEvent(event.getGuild().getIdLong(), modlogEventData);
+        }, ActionType.CHANNEL_OVERRIDE_UPDATE, ActionType.CHANNEL_UPDATE);
+    }
+
+    private String getCurrentPermissionStateForHolder(Permission permission, GuildChannel channel, IPermissionHolder holder) {
+        PermissionOverride permissionOverride = channel.getPermissionOverride(holder);
+        if (permissionOverride == null) {
+            return "unknown";
+        }
+
+        if (permissionOverride.getAllowed().contains(permission)) {
+            return "+";
+        } else if (permissionOverride.getInherit().contains(permission)) {
+            return "\\\\";
+        } else if (permissionOverride.getDenied().contains(permission)) {
+            return "x";
+        } else {
+            return "unknown";
+        }
+    }
```

Comment:
I know this is TODO still, but we probably want to make an enum for these values.